### PR TITLE
Updating README for local development

### PR DESCRIPTION
**Pull request by drmaas**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,4 +6,39 @@ A bakery for use by Spinnaker to produce machine images.
...
-It exposes a REST api which can be experimented with via the Swagger UI: http://localhost:8087/swagger/index.html
+It exposes a REST api which can be experimented with via the Swagger UI: http://localhost:8087/swagger-ui.html
+
+# Developing rosco
+
+Need to run rosco locally for development? Here's what you need to setup and run:
+
+## Environment Setup
+. git clone git@github.com:spinnaker/rosco.git
+. git clone https://github.com/spinnaker/spinnaker.git
```
Comment (duftler) [inline README.md]: Should probably use the same-style url's here.

**Comment by duftler** [comment]
The file doesn't render properly using github's View button. Please use github-flavored markdown to annotate code snippets and urls.

**Comment by duftler** [comment]
image

File: README.md (hunk 1 above)
Comment (drmaas) [inline README.md]: swagger-ui.html is the actual url it goes to.

File: README.md [hunk 2]
```diff
@@ -13,32 +13,48 @@ It exposes a REST api which can be experimented with via the Swagger UI: http://
 Need to run rosco locally for development? Here's what you need to setup and run:
 
 ## Environment Setup
-. git clone git@github.com:spinnaker/rosco.git
-. git clone https://github.com/spinnaker/spinnaker.git
+```
+git clone git@github.com:spinnaker/rosco.git
```
Comment (duftler) [inline README.md]: One url is `git@...` and one is `https://...`.

File: README.md [hunk 3]
```diff
@@ -6,4 +6,55 @@ A bakery for use by Spinnaker to produce machine images.
...
+
+## Verify redis
+```
+docker run -it --link redis:redis --rm redis redis-cli -h redis -p 6379
+(printf "PING\r\n";) | nc -v localhost 6379
+```
+
+## IDE setup
+```
+gradle wrapper
```
Comment (duftler) [inline README.md]: I think this line should start with a `#`.

File: README.md [hunk 4]
```diff
@@ -6,4 +6,55 @@ A bakery for use by Spinnaker to produce machine images.
...
+docker-compose up -d redis rush
+```
+
+## Verify redis
+```
+docker run -it --link redis:redis --rm redis redis-cli -h redis -p 6379
+(printf "PING\r\n";) | nc -v localhost 6379
+```
+
+# IDE setup
```
Comment (duftler) [inline README.md]: Sorry, what I meant was that the `gradle wrapper` line, which I think is a comment, should begin with a `#`.

File: README.md [hunk 5]
```diff
@@ -6,4 +6,55 @@ A bakery for use by Spinnaker to produce machine images.
...
+It exposes a REST api which can be experimented with via the Swagger UI: http://localhost:8087/swagger-ui.html
+
+# Developing rosco
+
+Need to run rosco locally for development? Here's what you need to setup and run:
+
+## Environment Setup
+```
+git clone git@github.com:spinnaker/rosco.git
+git clone https://github.com/spinnaker/spinnaker.git
```
Comment (duftler) [inline README.md]: What I meant here was that they should both be either `git clone git@` or `git clone https://`, but not one of each.

File: codestyle.xml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<code_scheme name="spinnaker">
```
Comment (duftler) [inline codestyle.xml]: If this formatting template is applied to the currently committed codebase, are changes made?

Comment (duftler) [inline codestyle.xml]: I'm not an expert on intellij, so I'm mainly looking to confirm that if somebody follows the 2 steps in 'Apply groovy code formatting scheme', that no reformatting immediately takes place.

Comment (drmaas) [inline codestyle.xml]: No - you still need to highlight the code you want and apply the reformat command.

**Comment by drmaas** [comment]
I found another gap in the documentation - I want to submit a request to Rush, however, since it runs a containerized packer build, it is not able to find my packer template (which is outside of the container).

How do you suggest getting this to work for local development?

**Comment by duftler** [comment]
LGTM.

**Comment by duftler** [comment]
Please squash the commits so I can merge.
This repo doesn't seem to have the 'Squash and merge commits' option enabled in github. I'll see about getting that enabled.

**Comment by duftler** [comment]
Rush relies on the script executor abstraction to run the jobs requested by rosco. There are two implementations, local and docker. Only the local implementation is supported. The docker artifacts are there for (not very compelling) historical reasons. It originally only supported running docker containers and we weren't sure when we were doing the initial refactoring if somebody would want to reenable that support for running non-bake jobs in pipelines (this was the initial domain of rush).

Now that there is solid support for running jobs via the kubernetes integration, this is no longer a concern. As part of the in-progress merging of rush into rosco all the deprecated ScriptExecutorDocker stuff is being dropped.

When you develop locally, you'll need to ensure that `packer` is available on the path (of rush for the moment, and of rosco shortly).

**Comment by duftler** [comment]
Btw, this ties back into our early conversation about the `configDir` config property in rosco. Rosco will pass that property on its requests to rush, and rush will need to be able to resolve that path when executing the packer commands.

**Pull request merged**